### PR TITLE
Non recursive untyped object deserializer

### PR DESCRIPTION
**Pull request by Niken-physics**
This is a work in progress moving the serializer from recursion to iteration. There are some issues: Still have not made the jsonarray non-recursive and also have removed the special cases of 1 or 2 elements. Only updated the vanilla parser. 

It uses same strategy as jsonnodeserializer to accomplish its goals

### CODE REVIEW

File: src/main/java/com/fasterxml/jackson/databind/deser/std/UntypedObjectDeserializer.java [hunk 1]
```diff
@@ -10,10 +10,13 @@
 import com.fasterxml.jackson.databind.deser.ContextualDeserializer;
 import com.fasterxml.jackson.databind.deser.ResolvableDeserializer;
 import com.fasterxml.jackson.databind.jsontype.TypeDeserializer;
+import com.fasterxml.jackson.databind.node.ArrayNode;
 import com.fasterxml.jackson.databind.type.LogicalType;
 import com.fasterxml.jackson.databind.type.TypeFactory;
 import com.fasterxml.jackson.databind.util.ClassUtil;
 import com.fasterxml.jackson.databind.util.ObjectBuffer;
+import jdk.internal.org.jline.reader.Parser;
+import sun.tools.jstat.Token;
```
Comment (cowtowncoder) [inline src/main/java/com/fasterxml/jackson/databind/deser/std/UntypedObjectDeserializer.java:19]: I don't know what these are but pretty sure we cannot use them (databind is Java 8, plus references to internal "sun.tools" are a big no no).

File: src/main/java/com/fasterxml/jackson/databind/deser/std/UntypedObjectDeserializer.java [hunk 2]
```diff
@@ -27,8 +30,8 @@
  */
 @JacksonStdImpl
 public class UntypedObjectDeserializer
-    extends StdDeserializer<Object>
-    implements ResolvableDeserializer, ContextualDeserializer
+        extends StdDeserializer<Object>
```
Comment (cowtowncoder) [inline src/main/java/com/fasterxml/jackson/databind/deser/std/UntypedObjectDeserializer.java:33]: Whitespace changes make this diff rather hard to read :-(

File: src/main/java/com/fasterxml/jackson/databind/deser/std/UntypedObjectDeserializer.java (hunk 1 above)
Comment (Niken-physics) [reply src/main/java/com/fasterxml/jackson/databind/deser/std/UntypedObjectDeserializer.java:19]: hmm probably got autoimported by my ide, I don't think it should be needed.  I will remove these

File: src/main/java/com/fasterxml/jackson/databind/deser/std/UntypedObjectDeserializer.java (hunk 2 above)
Comment (Niken-physics) [reply src/main/java/com/fasterxml/jackson/databind/deser/std/UntypedObjectDeserializer.java:33]: Yes sorry about that. Should I fix that and make another pr?

**Comment by cowtowncoder** [comment]
Ok, so couple of notes (some added inline):

* Overall makes sense (following same pattern as JsonNodeDeserializer)
* Would much prefer no changes to indentation, whitespace changes make code review hard due to bogus changes
* On optimizations; if we must 1 and 2 element cases, then we have to; may well be the case. 1 element much more important than 2.

**Comment by Niken-physics** [comment]
Will fix the above comments and see if I can do the optimization for the 1 element case as well. Sorry for some amateurish mistakes on my part!

**Comment by cowtowncoder** [comment]
@Niken-physics no problem at all! Really happy to get this contribution going, esp. if JSON Array case can be handled.

Oh, and forgot to mention: I think Vanilla one may be sufficient, although ideally general-purpose one can be solved too. But that's not the default so technically speaking I think Vanilla case is the only thing relevant for vulnerability aspect.

**Comment by Niken-physics** [comment]
Should have removed the not needed and bad dependencies and fixed at least most of the whitespace issues.

File: src/main/java/com/fasterxml/jackson/databind/deser/std/UntypedObjectDeserializer.java (hunk 2 above)
Comment (yawkat) [reply src/main/java/com/fasterxml/jackson/databind/deser/std/UntypedObjectDeserializer.java]: @Niken-physics if you fix it in this PR, the diff will update and it should be fine.

**Comment by cowtowncoder** [comment]
I ended up fixing this via #3473 which is sort of based on earlier `JsonNode` deserializer changes; closing this one.

Thank you for contributing this PR: it helped me figure out some aspects so it was helpful to get things resolved!

**Pull request closed without merging**